### PR TITLE
Ammend fund proposal

### PR DESCRIPTION
**Pull request by aalavanthan18**
Creating PR to pull in update project extrinsic changes

### CODE REVIEW

**Comment by samelamin** [comment]
LGTM, lets merge if CI passes

File: pallets/proposals/src/impls.rs [hunk 1]
```diff
@@ -101,6 +101,79 @@ impl<T: Config> Pallet<T> {
...
+    // Fill in the projects structure in advance
+    for milestone in proposed_milestones {
+        let milestone = Milestone { 
+            project_key,
+            milestone_key,
+            name: milestone.name.to_vec(),
+            percentage_to_unlock: milestone.percentage_to_unlock,
+            is_approved: false,
+        };
+        milestones.insert(milestone_key.clone(), milestone.clone());
```
Comment (f-gate) [inline pallets/proposals/src/impls.rs:142]: I dont think there is any need for these clones, u32 implements the Copy trait and milestone can be owned by the BtreeMap as it is not being used again in the scope.

Comment (f-gate) [reply pallets/proposals/src/impls.rs:142]: Reason for this is that clones are too expensive especically on a struct and copies are quick

File: pallets/proposals/src/lib.rs [hunk 2]
```diff
@@ -198,6 +198,13 @@ pub mod pallet {
             BalanceOf<T>,
             common_types::CurrencyId,
         ),
+        // Project has been updated
+        ProjectUpdated(
```
Comment (f-gate) [inline pallets/proposals/src/lib.rs:202]: What are we using this data for?

File: pallets/proposals/src/tests.rs [hunk 3]
```diff
@@ -2395,6 +2395,98 @@ fn test_finalise_milestone_is_ok_on_threshold_vote() {
...
+            project_key,
+            updated_project_name,
+            updated_project_logo,
+            updated_project_description,
+            updated_project_website,
+            updated_proposed_milestones.try_into().expect("Invalid proposed milestones"), 
+            updated_required_funds,
+            CurrencyId::Native,
+        ));
+
```
Comment (f-gate) [inline pallets/proposals/src/tests.rs:2478]: I see only an assertion on expected_project_name_in_event. Every property that should be changed by this extrinsic need affirmation.

File: pallets/proposals/src/weights.rs [hunk 4]
```diff
@@ -39,6 +40,11 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
             .saturating_add(T::DbWeight::get().reads(2_u64))
             .saturating_add(T::DbWeight::get().writes(2_u64))
     }
+    fn update_project() -> Weight {
```
Comment (f-gate) [inline pallets/proposals/src/weights.rs:43]: This needs a benchmark, writing static weights is not enough, that reminds me we need to regen weights.

**Review by f-gate — COMMENTED** [review]
Nice PR 
There's a few things id like to see done however, we are missing assertions in tests (maintaining integrity of a project) and the new extrinsic lacks a benchmark.

File: pallets/proposals/src/lib.rs (hunk 2 above)
Comment (aalavanthan18) [reply pallets/proposals/src/lib.rs:202]: This is an event definition that gets triggered at the end of the extrinsinc execution

File: pallets/proposals/src/tests.rs (hunk 3 above)
Comment (aalavanthan18) [reply pallets/proposals/src/tests.rs:2478]: True. That's definitely an input we can take to build additional tests associated to the update_project extrinsic.

File: pallets/proposals/src/impls.rs (hunk 1 above)
Comment (aalavanthan18) [reply pallets/proposals/src/impls.rs:142]: Not sure I understand this. I have followed the same convention as done within Create Project. Are you suggesting revamping clone to copy across both create and upate extrinsics?

File: pallets/proposals/src/weights.rs (hunk 4 above)
Comment (aalavanthan18) [reply pallets/proposals/src/weights.rs:43]: Agreed. Will work on this on a new PR

**Pull request merged**